components/familyCards/familyTree.tsx

```diff
@@ -0,0 +1,271 @@
+        firstName: "Alejandro",
+        lastName: "Santos",
+        relation: "Child",
+        dob: "08/05/2007",
+        dod: "",
+        spouse: {
+          firstName: "Lucia",
+          lastName: "Lopez",
+          relation: "Spouse",
+          dob: "09/10/2008",
+          dod: "",
+        },
+        children: [],
+      },
+      {
+        id: "3",
+        firstName: "Carmen",
+        lastName: "Santos",
+        relation: "Child",
+        dob: "09/12/2010",
+        dod: "",
+        children: [
+          {
+            id: "4",
+            firstName: "Nicolas",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "04/15/2015",
+            dod: "",
+            children: [],
+          },
+          {
+            id: "5",
+            firstName: "Renata",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "05/25/2016",
+            dod: "",
+            children: [],
+          },
+          {
+            id: "6",
+            firstName: "Alejandro",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "06/30/2017",
+            dod: "",
+            children: [],
+          },
+        ],
+      },
+    ],
+  });
+  const refs: { [key: string]: React.RefObject<HTMLDivElement> } = {};
+
+  const handleDrop = (droppedId: string, onId: string) => {
+    const updatedMembers = updateFamilyTree(familyMembers, droppedId, onId);
+
+    setFamilyMembers(updatedMembers);
+  };
+
+  const updateFamilyTree = (
+    members: FamilyMember,
+    droppedId: string,
+    onId: string
+  ) => {
+    const updatedMembers = JSON.parse(JSON.stringify(members));
+
+    const droppedMember = findMember(updatedMembers, droppedId);
+    const targetMember = findMember(updatedMembers, onId);
+
+    if (droppedMember && targetMember) {
+      const previousParent = findParent(updatedMembers, droppedId);
+      if (previousParent) {
+        previousParent.children = previousParent.children.filter(
+          (child) => child.id !== droppedId
+        );
+      }
+      const updatedDroppedMember = { ...droppedMember };
+
+      targetMember.children = [
+        ...(targetMember.children || []),
+        { ...droppedMember },
+      ];
+    }
+    return updatedMembers;
+  };
+
+  const findParent = (
+    member: FamilyMember,
+    childId: string
+  ): FamilyMember | null => {
+    for (const child of member.children || []) {
+      if (child.id === childId) return member;
+      const foundParent = findParent(child, childId);
+      if (foundParent) return foundParent;
+    }
+    return null;
+  };
+
+  const findMember = (
+    member: FamilyMember,
+    id: string
+  ): FamilyMember | null => {
+    if (member.id === id) return member;
+    for (const child of member.children || []) {
+      const found = findMember(child, id);
+      if (found) return found;
+    }
+    return null;
+  };
+  const getOrCreateRef = (id: string) => {
```

Comment:
Eslint is complaining about this. It is sayng that since you are using a hook, it should be inside of a component. I am not sure what the best practice is when it comes to that since Typescript skills are not the best.

![image](https://github.com/user-attachments/assets/1acd1d9f-5698-4f41-81a2-650570585709)

Reply:
Same thing here. Idk what's going on with my line selection